prboom2/src/hu_stuff.c

```diff
@@ -1249,6 +1249,16 @@ int HU_GetHealthColor(int health, int def)
 
 int HU_GetArmorColor(int armor, int def)
 {
+// [pull req] suggested change: armor color dictated by type (Advanced HUD)
+  if (plr->armortype >= 2)
+      return CR_BLUE;
+  else if (plr->armortype == 1)
+      return CR_GREEN;
+  else if (plr->armortype == 0)
+      return CR_RED;
+
+// [old behavior] armor color dictated by percent only, not type (Advanced HUD)
```

Comment:
I'd rather prefer if there was a switch to choose between "old" (i.e. by value) and "new" (i.e. by type) behaviour in the HUD menu. Not because I like menu switches so much, but because many players will have got used to the old behaviour and will get confused if this changes suddenly. Also, all the menu items concerning the armor color thresholds in the HUD menus would suddenly have no meaning anymore.

Reply:
Thank you for the feedback! I have implemented the new behavior as a toggle in the HUD menu in this latest commit. It is right above the armor color thresholds.